client/src/apps/Users/Users.tsx

```diff
@@ -0,0 +1,111 @@
+import { AppWindow } from '@components/AppWindow/AppWindow';
+import { DesktopIcon } from '@components/DesktopIcon/DesktopIcon';
+import { createSignal, createEffect, For } from 'solid-js';
+import { USER_MANAGER_ICON_INDEX } from '@components/AppIcon/IconSpritesheet';
+import styles from './Users.module.css';
+import { Input } from '@components/Input/Input';
+import { Button } from '@components/Button/Button';
+import { getUsers, postUser } from '../../services/users-service';
+import { User } from '../../types/user';
+
+/*
+ * TODO: This is a very basic app as my first contribution to the front.
+ *       We should replace it with correct user management component.
+ */
+function Users() {
+  const [showWindow, setShowWindow] = createSignal(false);
+  let buttonRef!: HTMLButtonElement;
+  const toggleWindow = () => {
+    setShowWindow((prev: boolean) => !prev);
+    if (buttonRef) {
+      buttonRef.blur();
+    }
+  };
+
+  const [error, setError] = createSignal('');
+  const [users, setUsers] = createSignal([]);
+
+  const onSubmit = async (e: Event) => {
+    setError('');
+    e.preventDefault();
+    const form = e.target as HTMLFormElement;
+    const formData = new FormData(form);
+
+    await postUser(formData)
+      .then((user) => setError(JSON.stringify(user)))
```

Comment:
I don't understand this - `setError` for success branch of the Promise? Why? Why with the JSON? Is that the leftover from debugging?

Reply:
Kind of, I copied setError from sound management. We just use it as visible output. It should be replaced system wise with something prettier. Maybe something  like [angular notification](https://www.telerik.com/kendo-angular-ui/components/notification#angular-notification-example)